tools/checklocks/state.go

```diff
@@ -18,23 +18,52 @@ import (
 	"fmt"
 	"go/token"
 	"go/types"
+	"sort"
 	"strings"
 	"sync/atomic"
-	"sort"
 
 	"golang.org/x/tools/go/ssa"
 )
 
 // lockType represents the current state of the lock.
 // The 0 value represents the unlocked state.
+// The valid states during analysis are:
+// unlocked, locked, deferredunlock and locked & deferredunlock
 type lockType int
 
 const (
-	unlocked lockType = iota
-	locked
-	// TODO(jamesyou): DEFERUNLOCK
+	unlocked lockType = 0
+	locked   lockType = 1 << iota
+	deferredunlock
 )
 
+// is checks if the lockType is in the state `t`.
+func (l lockType) is(t lockType) bool {
```

Comment:
"is" suggests ==. This is something more complex. If equality suffices, this is probably a better implementation. If equality does not suffice, I suggest changing the name and documenting the function more clearly.